File: platforms/kotlin/examples/core/texture_mip_chain/levels.kt
import org.fragmentcolor.*

val pixels = Array(8 * 8 * 4) { 0 }
val chain = TextureMipChain.prepare((
    pixels.asSlice(),
    TextureFormat.Rgba8UnormSrgb,
    arrayOf(8, 8),
))
val level_zero_bytes = chain.levels()[0]
val _ = level_zero_bytes

File: platforms/web/examples/core/texture_mip_chain/levels.js
import { TextureFormat, TextureMipChain } from "fragmentcolor";

const b64 = "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAADElEQVR4nGM4ceIEAAS0AlkWLoFAAAAAAElFTkSuQmCC";
const pngBytes = Uint8Array.from(atob(b64), c => c.charCodeAt(0));
const chain = TextureMipChain.prepare(pngBytes, TextureFormat.Rgba8UnormSrgb, null);
// level(0) returns the base mip level bytes
const level_zero_bytes = chain.level(0);